v4/core/base_service.go

```diff
@@ -180,11 +185,18 @@ func (service *BaseService) SetHTTPClient(client *http.Client) {
 }
 
 // DisableSSLVerification skips SSL verification.
+// This function sets a new http.Client instance on the service
+// and configures it to bypass verification of server certificates
+// and host names, making the client susceptible to "man-in-the-middle"
+// attacks.  This should be used only for testing.
 func (service *BaseService) DisableSSLVerification() {
```

Comment:
Because the client being used by the BaseService might now be a retryable client, we can't just replace the Transport field to disable ssl.  Instead, we create a new default client and then modify it's Transport's tls config.
The side-effect is that by disabling SSL, a new client instance is set on the BaseService, which sounds like an abrupt change, but we were previously replacing the Transport anyway so this is not much different now.

Reply:
Yes, at least for now, disable-ssl and retries are mutually exclusive.   I think there might be a way to support both simultaneously.   I can explore that as a follow-on.   For now, I'll document the current situation.